### DISCUSSION TITLE
Cache get keys by string pattern

### DISCUSSION THREAD

**Discussion by fazleyrabby** [laravel/framework#52174] [Ideas]
I was looking for a solution to find cache keys with string pattern like this: `cache()->keys("*bundle*")` to be able to clean the cached data that matches the string pattern.

Right now I am fetching the keys data by using Redis. Then looping through the data I am clearing the cache.

Here is my code snippet.
![CleanShot 2024-07-18 at 13 11 51@2x](https://github.com/user-attachments/assets/5ec29944-e746-43d7-bbee-1a8401d75205)

It would be really nice and quick to be able to get cache keys by using built in cache function.

**Comment by henzeb** [comment]
I had similar issues, thus I created https://packagist.org/packages/henzeb/laravel-cache-index a while back. 

you can also use tags. They are still available in Laravel
